Scripts- PowerShell and MS.Graph Module/W365 - List CloudPCs by Type.ps1

```diff
@@ -0,0 +1,83 @@
+﻿<#
+.COPYRIGHT
+Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+See LICENSE in the project root for license information.
+#>
+
+#Function to check if MS.Graph module is installed and is current version
+function invoke-graphmodule {
+    $graphavailable = (find-module -name microsoft.graph)
+    $vertemp = $graphavailable.version.ToString()
+    Write-Output "Latest version of Microsoft.Graph module is $vertemp" | out-host
+    $graphcurrent = (get-installedmodule -name Microsoft.Graph -ErrorAction SilentlyContinue) 
+
+    if ($graphcurrent -eq $null) {
+        write-output "Microsoft.Graph module is not installed. Installing..." | out-host
+        try {
+            Install-Module Microsoft.Graph -Force -ErrorAction Stop
+        }
+        catch {
+            write-output "Failed to install Microsoft.Graph Module" | out-host
+            write-output $_.Exception.Message | out-host
+            Return 1
+        }
+    }
+    $graphcurrent = (get-installedmodule -name Microsoft.Graph)
+    $vertemp = $graphcurrent.Version.ToString() 
+    write-output "Current installed version of Microsoft.Graph module is $vertemp" | out-host
+
+    if ($graphavailable.Version -gt $graphcurrent.Version) { write-host "There is an update to this module available." }
+    else
+    { write-output "The installed Microsoft.Graph module is up to date." | out-host }
+}
+
+#Function to connect to the MS.Graph PowerShell Enterprise App
+function connect-msgraph {
+
+    $tenant = get-mgcontext
+    if ($tenant.TenantId -eq $null) {
+        write-output "Not connected to MS Graph. Connecting..." | out-host
+        try {
+            Connect-MgGraph -Scopes "CloudPC.Read.All" -ErrorAction Stop | Out-Null
+        }
+        catch {
+            write-output "Failed to connect to MS Graph" | out-host
+            write-output $_.Exception.Message | out-host
+            Return 1
+        }   
+    }
+    $tenant = get-mgcontext
+    $text = "Tenant ID is " + $tenant.TenantId
+    Write-Output "Connected to Microsoft Graph" | out-host
+    Write-Output $text | out-host
+}
+#Function to set the profile to beta
+function set-profile {
+    Write-Output "Setting profile as beta..." | Out-Host
+    Select-MgProfile -Name beta
+}
+
+#Commands to load MS.Graph modules
+if (invoke-graphmodule -eq 1) {
+    write-output "Invoking Graph failed. Exiting..." | out-host
+    Return 1
+}
+
+#Command to connect to MS.Graph PowerShell app
+if (connect-msgraph -eq 1) {
+    write-output "Connecting to Graph failed. Exiting..." | out-host
+    Return 1
+}
+
+set-profile
+
+Write-Output " " | out-host
+
+$CloudPCs = Get-MgDeviceManagementVirtualEndpointCloudPC
+Write-Output "Enteprise SKU CloudPCs" | Out-Host
+foreach ($CloudPC in $CloudPCs) { if ($CloudPC.ServicePlanType -eq "enterprise") { Write-Output $CloudPC.Displayname | out-host } }
```

Comment:
AFAK, a tenant can only be enterprise tenant or VSB(business) tenant, and thus there should be only one type (enterprise or business) CloudPC under a tenant. I'm curious about what scenario does this script target for?

Reply:
I had a MVP friend who insisted you could have both business and enterprise coexist. The Enterprise would be in intue automatically, and you could enroll the Business SKU machines into Intune manually. 

In reality, I don't see this being a common scenario at all. If you think I should remove the script, I'm OK with that.